tools/deploy-api.ps1

```diff
@@ -0,0 +1,255 @@
+<#
+	.SYNOPSIS
+	Creates the file system folder structure along with IIS website and application for the given API.
+	.DESCRIPTION
+	Creates the file system folder structure for an API.
+	Creates the IIS website for the API.
+	Creates the application for the specified version of the API.
+	.PARAMETER site_name
+	The site name for the API. Required.
+	.PARAMETER app_name
+	The application name for the API. Required.
+	.PARAMETER app_version
+	The version of the API for the application. Required.
+	.PARAMETER site_hostname
+	The hostname of te API website. Required.
+	.PARAMETER code_folder
+	The folder from which to copy the latest code. Required.
+#>
+
+Param(
+	[Parameter(mandatory=$true)]
+	[string]$site_name = "",
+	
+	[Parameter(mandatory=$true)]
+	[string]$app_name = "",
+	
+	[Parameter(mandatory=$true)]
+	[string]$app_version = "", 
+	
+	[Parameter(mandatory=$true)]
+	[string]$site_hostname = "",
+	
+	[Parameter(mandatory=$true)]
+	[string]$code_folder = ""
+);
+
+## Named constants
+New-Variable -Name "CONTENT_ROOT" -Value "D:\Content"			-Option Constant;
+New-Variable -Name "SITES_ROOT"   -Value "D:\Content\APISites"	-Option Constant;
+New-Variable -Name "BACKUP_ROOT"  -Value "D:\backups"			-Option Constant;
+
+function Main() {
+	## The WebAdministration module requires elevated privileges.
+	$isAdmin = Is-Admin;
+	if( $isAdmin ) {
+		Write-Host -foregroundcolor 'green' "Starting...";
+		Import-Module WebAdministration;
+		CreateSite;
+	} else {
+		Write-Host -foregroundcolor 'red' "This script must be run from an account with elevated privileges.";
+	}
+}
+
+function Is-Admin {
+ $id = New-Object Security.Principal.WindowsPrincipal $([Security.Principal.WindowsIdentity]::GetCurrent());
+ $id.IsInRole([Security.Principal.WindowsBuiltinRole]::Administrator);
+}
+
+function CreateSite() {
+	## Check parameters
+	if ($site_name -eq "")

```

Comment:
Should we have a `function CheckParam([string]$param, [string]$ParamName)` to do these checks?